src/api/java/mekanism/api/math/FloatingLong.java

```diff
@@ -328,12 +332,81 @@ public FloatingLong divideEquals(FloatingLong toDivide) {
+            if (Long.compareUnsigned(toDivide.value, MAX_LONG_SHIFT) <= 0) { //don't case if *this* is < or > than shift
+                long div = toDivide.value * MAX_DECIMAL + toDivide.decimal;
+                return (Long.divideUnsigned(this.value, div) * MAX_DECIMAL) + (this.value % div * MAX_DECIMAL / div);
+            } else { //we already know todivide is > max_long_shift, and other case is impossible
+                if (Long.compareUnsigned(toDivide.value, Long.divideUnsigned(-1L, 2) + 1L) >= 0) //need to check anyways to avoid overflow on toDivide.value +1, so might as well return early
+                    return 1L;
+                if (Long.divideUnsigned(this.value, (toDivide.value + 1)) != Long.divideUnsigned(this.value, toDivide.value)) {
+                    // need to account for todivide.decimal in this case
+                    long q = Long.divideUnsigned(this.value, toDivide.value);
+                    return (toDivide.value * q + Long.divideUnsigned(toDivide.decimal * q, MAX_DECIMAL) <= this.value) ? q : q - 1;
+                }
+                else
+                    return Long.divideUnsigned(this.value, toDivide.value);
+            }
+        }
+        else {
+            // In this case, we're really multiplying (definitely need to account for decimal as well
+            if (Long.compareUnsigned(this.value, MAX_LONG_SHIFT) >= 0) {
+                return (this.value / toDivide.decimal) * MAX_DECIMAL //lose some precision here, have to add modulus
+                        + (this.value % toDivide.decimal) * MAX_DECIMAL / toDivide.decimal
+                        + this.decimal * MAX_DECIMAL / toDivide.decimal;
+            } else {
+                long d = this.value * MAX_DECIMAL;
+                return d / toDivide.decimal + this.decimal * MAX_DECIMAL / toDivide.decimal; //don't care about modulus since we're returning integers
+            }
+        }
+    }
+
+    /**
+     * Divides this {@link FloatingLong} by the given {@link FloatingLong} rounded down to an integer value, modifying the current object unless it is a constant in which case it instead returns the
+     * result in a new object. This gets clamped at the upper bound of {@link Integer#MAX_VALUE} rather than overflowing.
+     *
+     * @param toDivide The {@link FloatingLong} to divide by.
+     *
+     * @return The {@link FloatingLong} representing the value of dividing this {@link FloatingLong} by the given {@link FloatingLong}.
+     *
+     * @throws ArithmeticException if {@code toDivide} is zero.
+     * @apiNote It is recommended to set this to itself to reduce the chance of accidental calls if calling this on a constant {@link FloatingLong}
+     * <br>
+     * {@code value = value.divideToLong(toDivide)}
+     */
+    public int divideToInt(FloatingLong toDivide) {
+        return divideToLong(toDivide) <= (long)Integer.MAX_VALUE ? (int)divideToLong((toDivide)) : Integer.MAX_VALUE;
```

Comment:
This can just use `MathUtils#clampToInt`

Reply:
Fixed.